File: double-pull-delegate/src/main/java/zhan/pull/listener/OnStateChangeListener.java
package zhan.pull.listener;

import zhan.pull.ScrollState;


public interface OnStateChangeListener {

  void pullViewShow(ScrollState state);

  void pullViewHide(ScrollState state);

  void pullViewMove(ScrollState state, int offset);

  void pullViewOpenStart();

  void pullViewOpenFinish();
}

File: double-pull-delegate/src/main/java/zhan/pull/listener/OnStateChangeListener.kt
package zhan.pull.listener

import zhan.pull.ScrollState


interface OnStateChangeListener {

    fun pullViewShow(state: ScrollState)

    fun pullViewHide(state: ScrollState)

    fun pullViewMove(state: ScrollState, offset: Int)

    fun pullViewOpenStart()

    fun pullViewOpenFinish()
}
